### PR TITLE
[trainer] fix bug when batch size=1

### PR DESCRIPTION
**Pull request by LemonNoel**
### PR types

Bug fixes

### PR changes

APIs

### Description

- Add type checking in prediction_loop to avoid unexpected `logits[0]` when batch_size=1, which might lead to `Out of memory error`.

### CODE REVIEW

File: paddlenlp/trainer/trainer.py [hunk 1]
```diff
@@ -1952,19 +1797,11 @@ def prediction_step(
             return (loss, None, None)
 
         logits = nested_detach(logits)
-        if len(logits) == 1:
+        if isinstance(logits, (list, tuple)) and len(logits) == 1:
```
Comment (LemonNoel) [inline paddlenlp/trainer/trainer.py:1800]: 这里增加了类型检查，其他修改均为代码风格检查。

**Review by ZHUI — APPROVED** [review]
LGTM
